### PR TITLE
Update dependencies in the project #877

### PR DESCRIPTION
**Pull request by ANUKULJHA1506**
Updated all the dependencies, but the `extended_text_field` dependency needed to be downgraded because it required a higher SDK version than what was supported by the prescribed SDK levels.

### CODE REVIEW

**Review by adityastic — COMMENTED** [review]
Interesting. Are we talking about flutter sdk wrt to the downgrade or minimum android compilation sdk? I'm curious to know why 15 worked for us before

**Comment by adityastic** [comment]
@ANUKULJHA1506 It would be much better if you can setup dependabot instead which updates dependencies automatically rather than us periodically updating them

**Comment by ANUKULJHA1506** [comment]
@adityastic  i m  working on dependabot ,And well yes in my android studio it was saying to have higher sdk version to use this as dependency as when i was doing pub get , it was constantly showing errors, thats why i lower the dependency version.

**Comment by adityastic** [comment]
@ANUKULJHA1506 lets close this pr and create the changes for dependabot and then we can update them granularly

**Comment by ANUKULJHA1506** [comment]
@adityastic ok , is their any slack or any other platform where i can connect with the fossasia community members.

**Comment by adityastic** [comment]
@ANUKULJHA1506 indeed, there is a gitter channel in the README. Also can you remove pubspec changes. We'll get dependabot to make them. Thanks!

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+version: 2
+updates:
+  - package-ecosystem: "pub" # See documentation for possible values
+    directory: "/" # Location of package manifests
+    schedule:
+      interval: "weekly"
+    open-pull-requests-limit: 5  
+name: Auto-merge Dependabot PRs
+
+on:
```
Comment (adityastic) [inline .github/dependabot.yml]: Not sure why this is here. This is not dependabots config

**Review by adityastic — CHANGES_REQUESTED** [review]

File: .github/dependabot.yml (hunk 1 above)
Comment (ANUKULJHA1506) [reply .github/dependabot.yml]: these will send automatically PRs to update the versions , upto 5 PRs can be send at a time, then you can check them and push it in main branch.

Comment (ANUKULJHA1506) [reply .github/dependabot.yml]: do you want me to make some changes in depentabot config cause it is working fine in my forked repo.

Comment (adityastic) [reply .github/dependabot.yml]: There is a job configuration which is not relevant to dependabot.yml

Comment (adityastic) [reply .github/dependabot.yml]: I dont think it belongs here

File: pubspec.lock [hunk 2]
```diff
@@ -167,10 +167,10 @@ packages:
     dependency: "direct dev"
     description:
       name: flutter_lints
-      sha256: "9e8c3858111da373efc5aa341de011d9bd23e2c5c5e0c62bccf32438e192d7b1"
+      sha256: "3f41d009ba7172d5ff9be5f6e6e6abb4300e263aab8866d2a0842ed2a70f8f0c"
       url: "https://pub.dev"
     source: hosted
-    version: "3.0.2"
+    version: "4.0.0"
```
Comment (adityastic) [inline pubspec.lock:173]: We don't need changes in pubspec* files as dependabot will create PR's to update them. Can you remove both changes

**Review by adityastic — CHANGES_REQUESTED** [review]

File: pubspec.lock (hunk 2 above)
Comment (ANUKULJHA1506) [reply pubspec.lock:173]: ok , I'm looking into it.

**Comment by ANUKULJHA1506** [comment]
@adityastic is dependabot.yml need some changes or it is fine i have removed the  two jobs that u asked for, now dependabot will create only PR's to update them.

**Comment by adityastic** [comment]
@ANUKULJHA1506 both the pubspec file are still present

**Comment by ANUKULJHA1506** [comment]
@adityastic are you asking about dependabot.yml file or any other file ?

**Comment by adityastic** [comment]
@ANUKULJHA1506 please read the last comment

**Pull request closed without merging**